docs/writing_tests.rst

```diff
@@ -65,3 +69,40 @@ A more advanced example:
 
 
 The above YAML file contains all the possible options for a workflow test.
+
+Writing custom tests
+--------------------
+
+Pytest-workflow provides a way to run custom tests on files produced by a
+workflow.
+
+.. code-block:: python
+
+    import pathlib
+    import pytest
+
+    @pytest.mark.workflow(name='files containing numbers')
+    def test_div_by_three(workflow_dir):
+        number_file = workflow_dir / pathlib.Path("123.txt")
+
+        with number_file.open('rt') as file_h:
+            number_file_content = file_h.read()
+
+        assert int(number_file_content) % 3 == 0
+
+The ``@pytest.mark.workflow(name='files containing numbers')`` marks the test
+as belonging to a workflow named 'files containing numbers'. The mark can also
+be written without the explicit ``name`` key as ``@pytest.mark.workflow('files
+containing nummbers')``. This test will only run if the workflow 'files
+containing numbers' has run.
+
+``workflow_dir`` is a fixture. It does not work without a
+``pytest.mark.workflow('workflow_name')`` mark.  This is a
+`pathlib.Path <https://docs.python.org/3/library/pathlib.html>`_ object that
```

Comment:
Yes. But that will also clutter up the code example quite a bit, it looks a lot less clean. It might distract users from what is important.

It is a good suggestion, but when I wrote the type hint in the code it just looks so ugly:

```python
import pathlib
import pytest

@pytest.mark.workflow(name='files containing numbers')
def test_div_by_three(workflow_dir: pathlib.Path):
    number_file = workflow_dir / pathlib.Path("123.txt")

    with number_file.open('rt') as file_h:
        number_file_content = file_h.read()
    assert int(number_file_content) % 3 == 0
```